azure-pipelines.yml

```diff
@@ -150,9 +150,13 @@ steps:
     displayName: Install brew
 
   - script: |
+      curl -sSL https://get.haskellstack.org/ | sh
+      set -e
       curl -sSL https://raw.github.com/ndmitchell/hlint/master/misc/travis.sh | sh -s CI.hs
+      curl -sSL https://raw.github.com/ndmitchell/hlint/master/misc/travis.sh | sh -s ghc-lib-gen
       curl -sSL https://raw.github.com/ndmitchell/hlint/master/misc/travis.sh | sh -s examples/mini-hlint/src
+      curl -sSL https://raw.github.com/ndmitchell/hlint/master/misc/travis.sh | sh -s examples/strip-locs/src
       curl -sSL https://raw.github.com/ndmitchell/hlint/master/misc/travis.sh | sh -s examples/mini-compile/src
-      curl -sSL https://get.haskellstack.org/ | sh
-      stack --stack-yaml $(stack-yaml) --resolver $(resolver) setup > /dev/null
+      stack setup --stack-yaml $(stack-yaml) --resolver $(resolver) > /dev/null
+      stack exec --stack-yaml $(stack-yaml) --resolver $(resolver) --package extra --package optparse-applicative ghc -- -package extra -package optparse-applicative -Wall -Wno-name-shadowing -Werror -c CI.hs
```

Comment:
Aren't we running `CI.hs` twice now? Once here and once using Neil's `travis.sh` script?

Reply:
No, the travis script is running hlint over CI.hs - that's actually the change that has motivated all this. Then later we `stack exec ghc -- -c` CI.hs which checks it for warnings and errors and then finally we `stack runhaskell` to actually run it.